### DISCUSSION TITLE
Not able to find a protocol for FTS transfers

### DISCUSSION THREAD

**Discussion by arrabito** [DIRACGrid/DIRAC#7091] [Q&A] [answered]
Dear all,

It makes long time that I haven't used FTS for transfers.

I've just tried again to use it but I get the following error that seems to indicate that the transfer protocol is not found:

`2023-07-05 07:59:53 UTC DataManagement/FTS3Agent/treatOperation/1606283 ERROR: Could not submit FTS3Job FTS3Operation 1606283 : {'OK': False, 'Errno': 92, 'Message': 'Protocol not available ( 92 : Could not find a protocol )', 'CallStack': ['  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/threading.py", line 937, in _bootstrap\n    self._bootstrap_inner()\n', '  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/threading.py", line 980, in _bootstrap_inner\n    self.run()\n', '  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/threading.py", line 917, in run\n    self._target(*self._args, **self._kwargs)\n', '  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/multiprocessing/pool.py", line 125, in worker\n    result = (True, func(*args, **kwds))\n', '  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/site-packages/DIRAC/DataManagementSystem/Agent/FTS3Agent.py", line 469, in _treatOperation\n    res = ftsJob.submit(context=context, protocols=tpcProtocols)\n', '  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/site-packages/DIRAC/DataManagementSystem/Client/FTS3Job.py", line 699, in submit\n    res = self._constructTransferJob(pinTime, allLFNs, target_spacetoken, protocols=protocols)\n', '  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/site-packages/DIRAC/DataManagementSystem/Client/FTS3Job.py", line 348, in _constructTransferJob\n    res = dstSE.generateTransferURLsBetweenSEs(allLFNs, srcSE, protocols=protocols)\n', '  File "/opt/dirac/versions/v2.2.8-1680001240/Linux-x86_64/lib/python3.9/site-packages/DIRAC/Resources/Storage/StorageElement.py", line 842, in generateTransferURLsBetweenSEs\n    return S_ERROR(errno.ENOPROTOOPT, "Could not find a protocol ")\n']}`

Our configuration for Operations/<Setup>/ is:

![Capture d’écran 2023-07-05 à 10 23 53](https://github.com/DIRACGrid/DIRAC/assets/6996265/7fd6f112-1e07-4477-a40c-bb32e9044010)

For the Source SE:

![Capture d’écran 2023-07-05 à 10 25 03](https://github.com/DIRACGrid/DIRAC/assets/6996265/c1136fbe-ed76-42ed-ab3c-fc02bb2c865a)

and for the Target SE:

![Capture d’écran 2023-07-05 à 10 25 44](https://github.com/DIRACGrid/DIRAC/assets/6996265/5dfbc3ac-0d8a-4b37-909d-06f13171f5c4)

Is the problem related to the used protocols?

Thank you.

**Comment by chaen** [comment]
Hi,
we don't see the protocols of `CEA-Disk` so difficult to tell.
I created `dirac-dms-protocol-matrix` to debug exactly that sort of problems :-)

**Comment by arrabito** [comment]
Isn't it the section:

AccessProtocol.1/Protocol

that matters?

For CEA-Disk as you can see in the screenshot is:
AccessProtocol.1/Protocol=https

However I don't know for waht ProtocolsList is used?

Anyway using the command you suggested I get:

```
$ dirac-dms-protocol-matrix --FromSE=CEA-Disk --TargetSE=GRIF-Disk --FTS
Using sources: CEA-Disk
Using target: GRIF-Disk
Using the Virtual Organization: vo.cta.in2p3.fr
Could not generate transfer URLS src:CEA-Disk, dst:GRIF-Disk, error:Protocol not available ( 42 : Could not find a protocol )

```

So I guess that something is missing from my configuration. 

Any idea?

**Reply by chaen** [reply, marked as answer]
Yes, probably your protocol list in the operation section is incomplete, or you need a specific one for that SE.
See https://dirac.readthedocs.io/en/latest/AdministratorGuide/Resources/storage.html#multi-protocol
